pytorch_to_returnn/torch/nn/modules/rnn.py

```diff
@@ -29,16 +29,25 @@ class PackedSequence(PackedSequence_):
   always be satisfied.
   """
 
-  def get_batched_tensor(self) -> Tensor:
+  def get_batched_tensor(self, batch_first=None) -> Tensor:
     from .shape import FlattenBatch
     naming = Naming.get_instance()
     tensor_entry = naming.tensors[self.data]
     assert isinstance(tensor_entry, TensorEntry)
     for call_entry in tensor_entry.output_from_calls:
       if isinstance(call_entry.module.module, FlattenBatch):
-        return call_entry.orig_inputs_args[0]
+        out_tensor = call_entry.orig_inputs_args[0]
+        if batch_first is not None:
+          if batch_first != out_tensor.shape[0].is_batch_dim:
+            out_tensor = out_tensor.transpose(0, 1)
+          out_tensor_entry = naming.tensors[out_tensor]
+          if batch_first != out_tensor_entry.returnn_data.is_batch_major:
+            # batch and unflattened axis are always the first two, so just swap them
+            out_tensor_entry.returnn_axis_from_torch_axis[0] = 1
+            out_tensor_entry.returnn_axis_from_torch_axis[1] = 0
```

Comment:
I don't understand this.

First of all, the check is wrong. It never matters whether the RETURNN data is batch major. It matters whether the Torch tensor is batch major. But you already have that in your check above.

Then you cannot just reassign `returnn_axis_from_torch_axis`. Call `transpose` or so instead. But actually, again, this is just the code you already have above?

I think you don't need anything here anymore. Maybe just a check like:
```suggestion
          assert batch_first == out_tensor.shape[0].is_batch_dim
```

Reply:
Ah yes, you're right. I removed this now.